superset/security/manager.py

```diff
@@ -1017,9 +1017,9 @@ def clean_perms(self) -> None:
                 == None,  # noqa: E711
             )
         )
-        self.get_session.commit()
         if deleted_count := pvms.delete():
             logger.info("Deleted %i faulty permissions", deleted_count)
+        self.get_session.commit()
```

Comment:
I'm not really sure what's going on here. The commit was after a `SELECT` as opposed to the `DELETE`.